test/functional/sqlsrv/sqlsrv_ae_output_param_sqltype_numeric.phpt

```diff
@@ -107,17 +96,19 @@ function testOutputParam($conn, $spname, $direction, $dataType, $sqlType, $input
             }
         }
     } else {
-        $compatible = isCompatible($dataType, $sqlType);
+        $compatible = isCompatible($compatList, $dataType, $sqlType);
         if ($compatible && empty($errors)) {
             $success = true;
         } else {
             // Even if $dataType is compatible with $sqlType sometimes
-            // we still get errors from the server -- if so, it should
-            // return SQLSTATE '42000' or '22018', the former indicating
-            // an error when converting from one type to another and the
-            // latter indicating an incompatible conversion error
-            // Either one is acceptable for the purpose of this test
-            $success = ($errors[0]['SQLSTATE'] === '22018' || $errors[0]['SQLSTATE'] === '42000');
+            // we still get errors from the server -- if so, it might
+            // return either SQLSTATE '42000' or '22018' (operand type 
+            // clash but only happens with some certain types)
+            // E.g. when converting a bigint to int or an int to numeric, 
+            // SQLSTATE '42000' is returned, indicating an error when 
+            // converting from one type to another.

```

Comment:
I'm still not clear on why we would get errors if the types are supposed to be compatible. Do the errors occur only in AE?

Reply:
It only happens with non-AE, and there are many possible combinations -- will investigate further in another task